### DISCUSSION TITLE
Problem on set method of Column<T>

### DISCUSSION THREAD

**Discussion by JaschenDonri** [jtablesaw/tablesaw#1016] [Q&A]
Here is something I can't figure out with release 0.40.0:

when I'm using a specific column type, the set method works well, but when it's a generics column, the set method doesn't match the parameter.
I think the generics column set is very often used, can anyone tell me how to use it or provide a workaround other than the switch-case structure?

Following is the test case:

```
// works well
StringColumn test1 = StringColumn.create("test");
test1.set(i,test1,i-1);
test1.set(i,test1.get(i-1));

// doesn't work
Column<?> test2 = StringColumn.create("test");
test1.set(i,test2,i-1);
Object o = test2.get(i - 1);
```

**Comment by lwhite1** [comment]
In general, you should use the specific types. The generic types are there
to allow for the generally poor integration of Java primitives into the
world of Java generics. In tablesaw, using generics leads to auto-boxing of
entire columns, which is terrible for memory consumption and performance.

**Reply by lwhite1** [reply]
In looking at the example code again, i noticed a couple things:

```
// doesn't work
Column<?> test2 = StringColumn.create("test");
test1.set(i,test2,i-1);
Object o = test2.get(i - 1);
```
First, I would not expect this to work because you define test2 as Column<?>. The compiler doesn't know what kind of column it is and not all columns support set() with String arguments. It _might_ work if you try 

```
Column<String> test2....
```
Also, it looks like there's an error in the code: test1.set on the second line of the "doesn't work" part should be test2.set, I think.
